reactLab/src/InMemoryApp.js

```diff
@@ -0,0 +1,50 @@
+import {useState} from "react";
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import App from "./App.js";
+
+function InMemoryApp(props) {
+    const [data, setData] = useState(props.initialData);
+
+    function handleItemChanged(itemID, field, newValue){
+        function edit_helper(item) {
+            if (item.id === itemID){
+                item[field] = newValue;
```

Comment:
This isn't good because this is changing the contents of the data that is passed to it (so, when you do the data.map, you are changing a field within an item in data).
Instead of changing item, create a new object with all the old fields and values, but with the specified field updated to newValue.